### DISCUSSION TITLE
Discussion: annotations on the parameters of the record's primary constructor.

### DISCUSSION THREAD

**Discussion by Jason5Lee** [dotnet/csharplang#4064] [General] [answered]
Currently (.NET 5 rc2), the annotations on the parameters of the record's primary constructor only affect the constructor parameters but not the property. The decompilation of the following code

``` csharp
public sealed record Email([EmailAddress] string Address) {}
```

is

``` csharp
public sealed class Email : IContactInfo, IEquatable<Email>
{
	public string Address { get; init; }

	public Email([EmailAddress] string Address)
	{
		this.Address = Address;
	}

	// ...
}
```

which is not what I expect. I want something like

``` csharp
public sealed class EmailRecord : IContactInfo, IEquatable<EmailRecord>
{
	[EmailAddress]
	public string Address { get; }

	public EmailRecord(string address)
	{
		Address = address;
	}

       // ...
}
```

It's very useful when working with data validation in Model. I want an option to attach annotation on the property, or just attach the annotations on both constructor parameter and property.

**Comment by svick** [comment, marked as answer]
[You can do it like this:](https://sharplab.io/#v2:EYLgtghgzgLgpgJwD4AEBMBGAsAKFygZgAIo4IAbOAEyITgGMB7BGgUUgEtyAKAbQAcEjfohgBPEEXYQuAQSpU6UKAF0iKDAAYi8xXGUBKIgG8AvrnxopncrqVRZMGAg7AArvCKTHz1x7gm5ng4AHYQYPr8EPQBGgB0AEpuITAcEXEAwoxg/FyIAMqIAG4cMVC4xrhE1epWAJJQrAAe8Ahh5HUhHDCBuKZAA)

```c#
public sealed record Email([property: EmailAddress] string Address) {}
```
